### PR TITLE
Configuration changes + small optimization in EnergyValueRegistry, and many TODOs implemented.

### PR DESCRIPTION
**Pull request by MatanShahar**
1) I changed a bit the config "system", it now allows for generic enums for config values.
2) I added a check for the current config in `EnergyValueRegistry` to prevent unnecessary file reads.

### CODE REVIEW

File: src/main/java/com/pahimar/ee3/item/ItemEE.java [hunk 1]
```diff
@@ -41,4 +41,16 @@ protected String getUnwrappedUnlocalizedName(String unlocalizedName)
...
+
+    public static boolean IsAugmentable(Item item)
+    {
+        return item != null && item instanceof IAugmentable;
+
+    }
+
+    public static boolean IsAugment(Item item)
+    {
+        return item != null && item instanceof IAugment;
```
Comment (peddermaster2) [inline src/main/java/com/pahimar/ee3/item/ItemEE.java]: instanceof already does a null check

Comment (IntelliSun) [inline src/main/java/com/pahimar/ee3/item/ItemEE.java]: You're right, an old habit...

**Comment by utoc** [comment]
I tested your PR against my pack, I had a crash when loading.

[NullPointerException crash here](http://pastebin.com/LEe8fjbD) and the [official crash report which suspiciously leaves out the NPE](http://pastebin.com/snTV6B3J)

I've tested this with and without FastCraft, so that incompatibility can be ruled out.

**Comment by IntelliSun** [comment]
@utoc thanks for the notice, fixed with the latest merge.

**Comment by utoc** [comment]
Here is another although it may not be your fault since RWTema only just now implemented energy values in Extra Utilties and DynEMC has been broken since just a little before that.

Pastebin from console here: http://pastebin.com/YVpKA5qk
I did NOT get a crash-report from this.  Let me know if you want a full console output.

I removed Extra Utilities from the mod list and again crashed upon world load.
Console log from crash: http://pastebin.com/Xx5LhAjK
Crash-log in full: http://pastebin.com/escPXNUj

**Comment by IntelliSun** [comment]
Have you updated to f01d154c60329df6e449be59615b62db0fcc0b37? On my build everything is working fine...

**Comment by utoc** [comment]
I double checked  and I did properly checkout your latest update.  I tried again with a28d9df, verified in the source that the new e.StackTraces you put in were present and rebuilt.  I still cannot load the game with Extra Utilities, so once again I removed it from the modlist and got the following upon loading a world:

Console crash log: http://pastebin.com/pB78EVM9
Full crash log: http://pastebin.com/ZBXp5WLg

**Comment by pahimar** [comment]
Fairly out of sync with the current repo and design goals have changed. Closing.

**Pull request closed without merging**